coding/internal/file_data.cpp

```diff
@@ -35,7 +35,11 @@ FileData::FileData(string const & fileName, Op op)
 
   m_File = fopen(fileName.c_str(), modes[op]);
   if (m_File)
+  {
+    if (op == OP_APPEND)
```

Comment:
The comment says:
```
// Seek should not be called, if file is opened for append.
```

1. The seek operation is expensive, let's avoid it where it is not needed, by using ifdef.
2. Which tests were failing? Could it be that the tests were wrong? The MS documentation says:
```
When a file is opened by using the "a" access type or the "a+" access type, all write operations occur at the end of the file. The file pointer can be repositioned by using [fseek](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/fseek-fseeki64?view=msvc-170) or [rewind](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/rewind?view=msvc-170), but is always moved back to the end of the file before any write operation is performed. Therefore, existing data can't be overwritten.
```

Reply:
In `FilesContainerW::GetWriter`, `writer->Pos()` always returns 0 without this change.
> The seek operation is expensive, let's avoid it where it is not needed, by using ifdef.

It's probably noop on !windows since the file pointer is already at the end.

```
Running files_container_tests.cpp::FilesContainer_RewriteExisting
TID(1) ASSERT FAILED
coding\files_container.cpp:327
CHECK(m_info[i-1].m_offset < m_info[i].m_offset || m_info[i-1].m_size == 0 || m_info[i].m_size == 0)
Assertion failed: false, file coding\files_container.cpp, line 327
```

From MS docs (https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/ftell-ftelli64):
> If no I/O operation has yet occurred on a file opened for appending, the file position is the beginning of the file.

Reply:
Also, C17 7.21.3: https://www.open-std.org/jtc1/sc22/wg14/www/docs/n2310.pdf
> If a file can support positioning requests (such as a disk file, as opposed to a terminal), then a file position indicator associated with the stream is positioned at the start (character number zero) of the file, unless the file is opened with append mode in which case it is implementation-defined whether the file position indicator is initially positioned at the beginning or the end of the file.

Reply:
> Isn't it a system call?

Good point. Though I doubt it makes any noticeable difference since it's not in a hot loop.

> Are you speaking about this writer->Pos() call? 

Yes

> Shouldn't WritePaddingByPos() internally make a write that should update the pos accordingly?

No if `pos % kSectionAlignment == 0`.

Reply:
It would be great to make this patch explicit and with less impact. What about inserting

```
#ifdef _MSC_VER
// writer->Pos() below expects to return the end of the file, but it's not the case on Windows.
fseek....
#endif
m_info.emplace_back(tag, writer->Pos());
```

in the place of the failure?

Reply:
`FileData` is written with this behavior in mind (i.e. calling `Pos()` on a file opened in append mode returns the end of file). Adding conditional `fseek`s in this case is very fragile.